common/v2/features/BuyAssets/BuyAssetsForm.tsx

```diff
@@ -0,0 +1,199 @@
+import { EtherUUID, MOONPAY_ASSET_UUIDS } from 'v2/utils';
+import { ROUTE_PATHS, MOONPAY_API_QUERYSTRING, BUY_MYCRYPTO_WEBSITE } from 'v2/config';
+import { AccountDropdown, AssetDropdown, InlineMessage, ContentPanel } from 'v2/components';
+import { isAccountInNetwork } from 'v2/services/Store/Account/helpers';
+import { MoonpaySignerService } from 'v2/services/ApiService/MoonpaySigner';
+import { StoreContext } from 'v2/services/Store/StoreProvider';
+import { AssetContext } from 'v2/services/Store/Asset';
+
+const FormFieldItem = styled.fieldset`
+  margin-bottom: ${SPACING.LG};
+`;
+
+const FormFieldLabel = styled.label`
+  display: flex;
+  font-size: 1rem;
+  margin-bottom: ${SPACING.SM};
+  font-weight: 400;
+  align-items: center;
+  flex-wrap: wrap;
+`;
+
+const FormFieldSubmitButton = styled(Button)`
+  width: 100%;
+  display: inline-block;
+  margin-bottom: 15px;
+  &:disabled {
+    background-color: rgba(0, 122, 153, 0.3);
+  }
+`;
+
+const NavigationWarning = styled.div`
+  color: ${COLORS.GREY};
+  margin-bottom: ${SPACING.SM};
+  display: flex;
+  justify-content: center;
+`;
+
+const WarningIcon = styled(Icon)`
+  margin-right: 5px;
+  vertical-align: middle;
+
+  svg {
+    color: ${COLORS.GREY};
+  }
+`;
+
+const Divider = styled.div`
+  height: 1px;
+  margin-bottom: 20px;
+  background: #e3edff;
+`;
+
+interface IBuyFormState {
+  account: StoreAccount;
+  asset: Asset;
+}
+
+enum SubmissionType {
+  SEND_TO_SELF = 'SEND_TO_SELF',
+  SEND_TO_OTHER = 'SEND_TO_OTHER'
+}
+
+export const BuyAssetsForm = ({}: any) => {
+  const history = useHistory();
+  const { accounts } = useContext(StoreContext);
+  const { assets } = useContext(AssetContext);
+  const ethAsset = assets.find(asset => asset.uuid === EtherUUID) as Asset;
+
+  const initialFormikValues: IBuyFormState = {
+    account: {} as StoreAccount,
+    asset: ethAsset
+  };
+
+  const BuyFormSchema = Yup.object().shape({
+    account: Yup.object().required(translateRaw('REQUIRED')),
+    asset: Yup.object().required(translateRaw('REQUIRED'))
+  });
+
+  const filteredAssets = assets.filter(({ uuid }) => MOONPAY_ASSET_UUIDS.includes(uuid));
+
+  const handleSubmission = (values: IBuyFormState, typeOfSubmission: SubmissionType) => {
+    if (typeOfSubmission === SubmissionType.SEND_TO_SELF) {
+      const urlQuery = `${MOONPAY_API_QUERYSTRING}&currencyCode=${values.asset.ticker}&walletAddress=${values.account.address}`;
+
+      MoonpaySignerService.instance
+        .signUrlQuery(urlQuery)
+        .then((signature: any) => {
+          const redirectQueryParams = `?currencyCode=${values.asset.ticker}&walletAddress=${
+            values.account.address
+          }&signature=${encodeURIComponent(signature)}`;
+          window.open(`${BUY_MYCRYPTO_WEBSITE}${redirectQueryParams}`, '_blank');
+        })
+        .catch(err => {
+          console.debug('err detected: ', err);
+          window.open(`${BUY_MYCRYPTO_WEBSITE}`, '_blank');
```

Comment:
Any reason we open the window also if something goes wrong?

Reply:
We're assuming the user still wants to continue on, the moonpay widget handles the lack of data and since it'll show up at the end of the flow, it should be no-biggie.